lib/core/CStatistics.cc

```diff
@@ -67,23 +67,31 @@ CStat& CStatistics::stat(int index) {
     return ms_Instance.m_Stats[index];
 }
 
+void CStatistics::cacheStats() {
```

Comment:
Is there ever a valid situation when we expect this to be called when `m_Cache` is not null?  If not, it would be good to log a warning as the first operation if `m_Cache != nullptr`.